sw/device/silicon_creator/lib/shutdown_unittest.cc

```diff
@@ -0,0 +1,399 @@
+            {0, 10, 10, 0xFFFFFFFF},  // Class B
+            {0, 0, 0, 0},             // Class C
+            {0, 0, 0, 0},             // Class D
+        },
+};
+
+constexpr DefaultAlertClassification default_alert_classification[] = {
+    ALERTS(FULL)};
+
+static_assert(
+    ARRAYSIZE(default_alert_classification) == kTopEarlgreyAlertIdLast + 1,
+    "The default alert classification doesn't match the number of alerts");
+
+constexpr alert_class_t kClasses[] = {
+    kAlertClassA,
+    kAlertClassB,
+    kAlertClassC,
+    kAlertClassD,
+};
+
+alert_enable_t RomAlertClassEnable(alert_class_t cls) {
+  // Note: these need to match with `rom_alert_class_en` above.
+  switch (cls) {
+    case kAlertClassA:
+      return kAlertEnableLocked;
+    case kAlertClassB:
+      return kAlertEnableEnabled;
+    case kAlertClassC:
+      return kAlertEnableNone;
+    case kAlertClassD:
+      return kAlertEnableNone;
+    // Class X (and all other invalid classes) default to class A's enable
+    // status.
+    default:
+      return kAlertEnableLocked;
+  }
+}
+
+alert_escalate_t RomAlertClassEscalation(alert_class_t cls) {
+  // Note: these need to match with `rom_alert_class_escalation` above.
+  switch (cls) {
+    case kAlertClassA:
+      return kAlertEscalatePhase3;
+    case kAlertClassB:
+      return kAlertEscalatePhase3;
+    case kAlertClassC:
+      return kAlertEscalateNone;
+    case kAlertClassD:
+      return kAlertEscalateNone;
+    // Class X (and all other invalid classes) default to class A's escalate
+    // setting.
+    default:
+      return kAlertEscalatePhase3;
+  }
+}
+
+class ShutdownTest : public mask_rom_test::MaskRomTest {
+ protected:
+  OtpConfiguration otp_config_ = otp_config;
+  MockOtp otp_;
+  MockShutdown shutdown_;
+  mask_rom_test::MockAlert alert_;
+
+  void SetupOtpReads() {
+    // Make OTP reads retrieve their values from `otp_config_`.
+    ON_CALL(otp_, otp_read32(testing::_))
+        .WillByDefault([this](uint32_t address) {
+          // Must be aligned and in the SW_CFG partition.
+          EXPECT_EQ(address % 4, 0);
+          EXPECT_GE(address, OTP_CTRL_PARAM_OWNER_SW_CFG_OFFSET);
+          EXPECT_LT(address, OTP_CTRL_PARAM_OWNER_SW_CFG_OFFSET +
+                                 sizeof(this->otp_config_));
+          // Convert the address to a word index.
+          uint32_t index = (address - OTP_CTRL_PARAM_OWNER_SW_CFG_OFFSET) / 4;
+          const uint32_t *words =
+              reinterpret_cast<const uint32_t *>(&this->otp_config_);
+          return words[index];
+        });
+  }
+
+  void ExpectClassConfigure() {
+    ExpectClassConfigure(0);
+    ExpectClassConfigure(1);
+    ExpectClassConfigure(2);
+    ExpectClassConfigure(3);
+  }
+
+  void ExpectClassConfigure(size_t i) {
+    alert_class_t klass = kClasses[i];
+    EXPECT_CALL(alert_, alert_class_configure(klass, testing::_))
+        .WillOnce(Invoke([this, klass, i](alert_class_t cls,
```

Comment:
Why capture `klass` if you're already capturing `i`?

Reply:
To avoid re-computing it.  Fixed.